openmetadata-service/src/test/java/org/openmetadata/service/alerts/AlertsRuleEvaluatorTest.java

```diff
@@ -3,138 +3,165 @@
-            .resource("table")
-            .entity(table)
-            .entityRepository(Mockito.mock(TableRepository.class))
-            .build();
-    // SubjectContext subjectContext = new SubjectContext(user);
-    // RuleEvaluator ruleEvaluator = new RuleEvaluator(null, subjectContext, resourceContext);
-    ChangeEvent event = new ChangeEvent();
-    // evaluationContext = new StandardEvaluationContext(ruleEvaluator);
+  public static void setup(TestInfo test) throws URISyntaxException, IOException {
+    tableResourceTest = new TableResourceTest();
+    tableResourceTest.setup(test);
   }
 
   @Test
-  void test_noOwner() {
-    // Set no owner to the entity and test noOwner method
-    table.setOwner(null);
-    assertTrue(evaluateExpression("noOwner()"));
-    assertFalse(evaluateExpression("!noOwner()"));
-
-    // Set owner to the entity and test noOwner method
-    table.setOwner(new EntityReference().withId(UUID.randomUUID()).withType(Entity.USER));
-    assertFalse(evaluateExpression("noOwner()"));
-    assertTrue(evaluateExpression("!noOwner()"));
+  void test_matchAnySource() {
+    // Create a change Event
+    ChangeEvent changeEvent = new ChangeEvent();
+    changeEvent.setEntityType("alert");
+    AlertsRuleEvaluator alertsRuleEvaluator = new AlertsRuleEvaluator(changeEvent);
+    EvaluationContext evaluationContext = new StandardEvaluationContext(alertsRuleEvaluator);
+    assertTrue(evaluateExpression("matchAnySource('alert')", evaluationContext));
+    assertFalse(evaluateExpression("matchAnySource('bot')", evaluationContext));
   }
 
   @Test
-  void test_isOwner() {
-    // Table owner is a different user (random ID) and hence isOwner returns false
-    table.setOwner(new EntityReference().withId(UUID.randomUUID()).withType(Entity.USER).withName("otherUser"));
-    assertFalse(evaluateExpression("isOwner()"));
-    assertTrue(evaluateExpression("!isOwner()"));
-
-    // Table owner is same as the user in subjectContext and hence isOwner returns true
-    table.setOwner(new EntityReference().withId(user.getId()).withType(Entity.USER).withName(user.getName()));
-    assertTrue(evaluateExpression("isOwner()"));
-    assertFalse(evaluateExpression("!isOwner()"));
-
-    // noOwner() || isOwner() - with noOwner being true and isOwner false
-    table.setOwner(null);
-    assertTrue(evaluateExpression("noOwner() || isOwner()"));
-    assertFalse(evaluateExpression("!noOwner() && !isOwner()"));
-
-    // noOwner() || isOwner() - with noOwner is false and isOwner true
-    table.setOwner(new EntityReference().withId(user.getId()).withType(Entity.USER).withName(user.getName()));
-    assertTrue(evaluateExpression("noOwner() || isOwner()"));
-    assertFalse(evaluateExpression("!noOwner() && !isOwner()"));
+  void test_matchAnyOwnerName(TestInfo test) throws IOException {
+    // Create Table
+    List<Column> columns = List.of(new Column().withName("c1").withDataType(ColumnDataType.INT));
+    CreateTable create =
```

Comment:
Use TableResourceTest.getColumn static method to create columns

Reply:
updated